src/vivarium_cluster_tools/vadmin/repositories.py

```diff
@@ -0,0 +1,230 @@
+    """Creates a new remote repository.
+
+    Parameters
+    ----------
+    service
+        Either 'stash' or 'github'.  The service to generate the new repository
+        on.
+    oauth_token
+        The user's credentials to access the service.
+    repo_name
+        The name of the new remote repository to create.
+
+    Returns
+    -------
+        The ssh url that the repository can be cloned from.
+
+    """
+    if service == 'stash':
+        headers = {'Content-Type': 'application/json', 'Authorization': f'Bearer {oauth_token}'}
+        api_endpoint = 'https://stash.ihme.washington.edu/rest/api/1.0/projects/CSTE/repos'
+        payload = {'name': repo_name,
+                   'scmId': 'git'}
+    else:  # service == 'github'
+        headers = {'Authorization': f'token {oauth_token}'}
+        api_endpoint = 'https://api.github.com/orgs/ihmeuw/repos'
+        payload = {'name': repo_name}
+
+    logger.debug(f'Creating the repository {repo_name} on {service}.')
+    response = requests.post(api_endpoint, headers=headers, data=json.dumps(payload))
+    content = parse_repo_creation_response(response)
+
+    if service == 'stash':
+        clone_ref = [ref['href'] for ref in response.json()['links']['clone'] if ref['name'] == 'ssh'].pop()
+    else:  # service == github
+        clone_ref = content['ssh_url']
+
+    return clone_ref
+
+
+def delete_repository(service: str, oauth_token: str, repo_name: str):
+    """Deletes a remote repository.
+
+    Parameters
+    ----------
+    service
+        Either 'stash' or 'github'.  The service to delete the repository
+        from.
+    oauth_token
+        The user's credentials to access the service.
+    repo_name
+        The name of the remote repository to delete.
+
+    """
+    if service == 'stash':
+        headers = {'Content-Type': 'application/json', 'Authorization': f'Bearer {oauth_token}'}
+        api_endpoint = f'https://stash.ihme.washington.edu/rest/api/1.0/projects/CSTE/repos/{repo_name}'
+    else:  # service == github
+        headers = {'Authorization': f'token {oauth_token}'}
+        api_endpoint = f'https://api.github.com/repos/ihmeuw/{repo_name}'
+
+    logger.debug(f'Deleting the repository {repo_name} on {service}.')
+    response = requests.delete(api_endpoint, headers=headers)
+    parse_repo_deletion_response(response)
+
+
+def parse_repo_creation_response(response: requests.Response) -> dict:
+    """Parses the response from repository creation.
+
+    Parameters
+    ----------
+    response
+        The server response from 'stash' or 'github' after requesting to
+        create a new repository.
+
+    Returns
+    -------
+        The content of the server response.
+
+    Raises
+    ------
+    HTTPError
+        If the repository creation was unsuccessful.
+
+    """
+    if response.status_code == 201:
+        logger.info('Repository successfully created')
+        return response.json()
+    elif response.status_code == 409:
```

Comment:
this seems odd to raise a HTTPError here - the communication wasn't the cause of the error?